pkg/kritis/admission/admission.go

```diff
@@ -57,88 +62,190 @@ var (
 	defaultViolationStrategy = violation.LoggingStrategy{}
 )
 
-// This admission controller looks for the breakglass annotation
-// If one is not found, it validates against image security policies
-// TODO: Check for attestations
+var (
+	runtimeScheme = runtime.NewScheme()
+	codecs        = serializer.NewCodecFactory(runtimeScheme)
+)
+
 func AdmissionReviewHandler(w http.ResponseWriter, r *http.Request) {
-	glog.Info("Starting admission review handler version %s ...", version.Commit)
-	pod, err := admissionConfig.retrievePod(r)
+	glog.Infof("Starting admission review handler, version: %s",
+		version.Commit)
+	body, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		glog.Error(err)
-		w.WriteHeader(http.StatusBadRequest)
+		glog.Errorf("Error reading body: %v", err)
+		http.Error(w, "can't read body", http.StatusBadRequest)
 		return
 	}
-	// First, check for a breakglass annotation on the pod
-	if checkBreakglass(pod) {
-		glog.Infof("found breakglass annotation, returning successful status")
-		returnStatus(constants.SuccessStatus, constants.SuccessMessage, w)
-		return
+
+	ar := v1beta1.AdmissionReview{}
+	deserializer := codecs.UniversalDeserializer()
```

Comment:
Can we wrap this in a function and return a list of images?

Reply:
We return only an image list as the breakglass checks check higher level metadata Pod.annotation, Deployment.annotation so we need the functions to be at that level